### PR TITLE
e2e:serial: deflake podscope failures

### PR DESCRIPTION
**Pull request by Tal-or**
A couple of improvements for the serial e2e tests.
Some are intended to solve actual failures and some are just best-effort tuning.

The tests were performed on a cluster with nodes configured with a pod scope policy.

### CODE REVIEW

File: test/e2e/serial/tests/sched_removal.go [hunk 1]
```diff
@@ -137,10 +139,29 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources scheduler restar
 		It("[case:1][test_id:48069][tier2] should schedule any pending workloads submitted while the scheduler was unavailable", func() {
 			var err error
 
+			dps, err := getDPOwnedBy(fxt.Client, nroSchedObj.ObjectMeta)
```
Comment (fromanirh) [inline test/e2e/serial/tests/sched_removal.go:142]: this is fine. We can also use the scheduler object status field.

File: test/e2e/serial/tests/sched_removal.go [hunk 2]
```diff
@@ -137,10 +139,29 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources scheduler restar
...
 			By(fmt.Sprintf("deleting the NRO Scheduler object: %s", nroSchedObj.Name))
 			err = fxt.Client.Delete(context.TODO(), nroSchedObj)
 			Expect(err).ToNot(HaveOccurred())
 
+			// make sure scheduler deployment is gone
+			Eventually(func() bool {
+				for _, dp := range dps {
+					if err := fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(dp), dp); err != nil {
+						if apierrors.IsNotFound(err) {
+							return true
```
Comment (fromanirh) [inline test/e2e/serial/tests/sched_removal.go:154]: it seems to me this means we bail out as soon as the first deployment is gone - but we should actually wait for the _last_ to be gone (meaning all of them are gone), right?

**Review by ffromani — APPROVED** [review]
/lgtm
all the changes look reasonnable to me

File: test/e2e/serial/tests/sched_removal.go (hunk 2 above)
Comment (Tal-or) [reply test/e2e/serial/tests/sched_removal.go:154]: We should only have a single deployment for the scheduler.

We can actually assert there is only one and failed the test if it's different.

Comment (fromanirh) [reply test/e2e/serial/tests/sched_removal.go:154]: I'd prefer this way. `getDPOwnedBy` should either return 1 Deployment (success) or nil (on error).

**Comment by fromanirh** [comment]
plase rebase on top of main branch

**Comment by Tal-or** [comment]
/retest
looks like the ci stuck

**Pull request merged**